src/CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+include("list_of_files.cmake")
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_Fortran_FLAGS "-g -auto -convert big_endian -assume byterecl -fp-model strict -fpp")
+  set(fortran_d_flags "-i4 -r8")
+  set(fortran_8_flags "-i8 -r8")
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
+  set(CMAKE_Fortran_FLAGS "-g -fconvert=big-endian -cpp")
+  set(CMAKE_Fortran_FLAGS_DEBUG "-ggdb -Wall")
+  set(fortran_d_flags "-fdefault-real-8")
+  set(fortran_8_flags "-fdefault-integer-8 -fdefault-real-8")
+endif()
+
+set(kinds "4" "8" "d")
+foreach(kind ${kinds})
+  set(lib_name ${PROJECT_NAME}_${kind})
+  set(module_dir "${CMAKE_CURRENT_BINARY_DIR}/include_${kind}")
+
+  # Compiled with preprocessor definition LSIZE=D, not d
+  string(TOUPPER ${kind} kind_definition)
+
+  set(BUILD_FLAGS "${fortran_${kind}_flags}")
+
+  add_library(${lib_name} STATIC ${fortran_src})
+  add_library(${PROJECT_NAME}::${lib_name} ALIAS ${lib_name})
+
+  target_compile_definitions(${lib_name} PRIVATE "LSIZE=${kind_definition}")
+  set_target_properties(${lib_name} PROPERTIES COMPILE_FLAGS "${BUILD_FLAGS}")
+  set_target_properties(${lib_name} PROPERTIES Fortran_MODULE_DIRECTORY "${module_dir}")
+
+  if(OpenMP_Fortran_FOUND)
+    target_link_libraries(${lib_name} PUBLIC OpenMP::OpenMP_Fortran)
+  endif()
+
+  target_link_libraries(${lib_name} PUBLIC sp::sp_${kind})
+  
+  target_include_directories(${lib_name}
+    PUBLIC $<BUILD_INTERFACE:${module_dir}>
+    $<INSTALL_INTERFACE:include_${kind}>)
+
+  list(APPEND LIB_TARGETS ${lib_name})
+  
+  install(DIRECTORY ${module_dir} DESTINATION ${CMAKE_INSTALL_PREFIX})
+  install(FILES iplib_${kind}.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include_${kind} RENAME iplib.h)
```

Comment:
If this library is called ‘ip2’, why is the header ‘iplib.h’?

Reply:
That is a good point.  ip2lib was created from iplib.  I must have forgot to change that.  @kgerheiser can we rename them?